### PR TITLE
34 weatherdata

### LINKED ISSUE #34: [BPS] [HVAC] [CFD] Weatherdata
We need a database for weatherdata or an API to use existing databases to include them later into our Modelica models.
DWD provides weather data as .dat here: https://kunden.dwd.de/obt/

* For AixLib we use .mos weather data and have a converter.
* What format is used for HKESim?
* What format does EnergyPlus use?
* Is weather data relevant for CFD?

Todo:
create a new task in common that has city, format and target path as input

### PR DESCRIPTION
**Pull request by DaJansenGit**
## Changes:
- add mandatory option for `sim_settings`
- add `ifc_attr_name` as option for attribute system. This is for "real" attributes. Needed in this branch because longitude and latitude of IfcSite are attributes and not property sets
- Weather files are now a mandatory `sim_setting` and have no default value
- Preparations for IFC data based automatic download of DWD TRY files. (waiting for repsone from DWD)
- Add tests for weather files
- Weather task is not different anymore for TEASER and EnergyPlus but only one task
- Adjusted all tests and examples with weather files. Also needed for LCA right now even if no weather file is needed. This needs to be fixed for `sim_settings` in #583 

closes #34

### CODE REVIEW

**Review by veronikarichter — CHANGES_REQUESTED** [review]
Some minor modifications required, looks good in general.

File: bim2sim/tasks/common/weather.py [hunk 1]
```diff
@@ -1,50 +1,109 @@
...
+        weather_file = None
+        # try to get weather file from settings
+        if self.playground.sim_settings.weather_file_path:
+            weather_file = self.playground.sim_settings.weather_file_path
+        # try to get TRY weather file for location of IFC
+        if not weather_file:
+            raise NotImplementedError("Waiting for response from DWD if we can"
+                                      "implement this")
+            # lat, long = self.get_location_lat_long_from_ifc(instances)
+            # weather_file = self.get_weatherfile_from_dwd(lat, long)
```
[the remark this answers is not in the document]
Comment (DaJansenGit) [reply bim2sim/tasks/common/weather.py:21]: Only germany, I will do when/if we implement it.

File: test/unit/tasks/common/test_inspect.py [hunk 2]
```diff
@@ -61,10 +62,13 @@ def test_case_2(self):
...
         self.test_dir = tempfile.TemporaryDirectory()
         ifc_paths = {
-            IFCDomain.hydraulic: sample_root /
+            IFCDomain.hydraulic: test_rsrc_path /
+                                 'hydraulic/ifc/'
                                  'B01_3_HeatExchanger_noPorts.ifc'}
         self.project = Project.create(self.test_dir.name, ifc_paths,
                                       plugin=PluginDummy, )
+        self.project.sim_settings.weather_file_path = (
+                test_rsrc_path / 'weather_files/DEU_NW_Aachen.105010_TMYx.epw')
```
Comment (veronikarichter) [inline test/unit/tasks/common/test_inspect.py:71]:
```suggestion
                test_rsrc_path / 'weather_files/DEU_NW_Aachen.105010_TMYx.mos')
```

File: test/unit/tasks/common/test_inspect.py [hunk 3]
```diff
@@ -79,10 +83,13 @@ def test_case_3(self):
...
         self.test_dir = tempfile.TemporaryDirectory()
         ifc_paths = {
-            IFCDomain.hydraulic: sample_root /
+            IFCDomain.hydraulic: test_rsrc_path /
+                                 'hydraulic/ifc/'
                                  'B01_4_HeatExchanger_noConnection.ifc'}
         self.project = Project.create(self.test_dir.name, ifc_paths,
                                  plugin=PluginDummy, )
+        self.project.sim_settings.weather_file_path = (
+                test_rsrc_path / 'weather_files/DEU_NW_Aachen.105010_TMYx.epw')
```
Comment (veronikarichter) [inline test/unit/tasks/common/test_inspect.py:92]:
```suggestion
                test_rsrc_path / 'weather_files/DEU_NW_Aachen.105010_TMYx.mos')
```

File: test/unit/tasks/common/test_inspect.py [hunk 4]
```diff
@@ -95,10 +102,13 @@ def test_case_4(self):
...
         self.test_dir = tempfile.TemporaryDirectory()
         ifc_paths = {
-            IFCDomain.hydraulic: sample_root /
+            IFCDomain.hydraulic: test_rsrc_path /
+                                 'hydraulic/ifc/'
                                  'B01_5_HeatExchanger_mixConnection.ifc'}
         self.project = Project.create(self.test_dir.name, ifc_paths,
                                  plugin=PluginDummy, )
+        self.project.sim_settings.weather_file_path = (
+                test_rsrc_path / 'weather_files/DEU_NW_Aachen.105010_TMYx.epw')
```
Comment (veronikarichter) [inline test/unit/tasks/common/test_inspect.py:111]:
```suggestion
                test_rsrc_path / 'weather_files/DEU_NW_Aachen.105010_TMYx.mos')
```

**Review by veronikarichter — CHANGES_REQUESTED** [review]

**Review by veronikarichter — APPROVED** [review]

**Pull request merged**